R/filter_hierarchical.R

```diff
@@ -4,38 +4,58 @@
 #' expression. Rows can be filtered on the values of any of the possible statistics (`n`, `p`, and `N`) provided they
-#' are included at least once in the table, as well as the values of any `by` variables. Filtering is only applied to
-#' rows that correspond to the innermost variable in the hierarchy - all outer variable (summary) rows are kept
-#' regardless of whether they meet the filtering criteria themselves. In addition to filtering on individual statistic
-#' values, filters can be applied across the row (i.e. across all `by` variable values) by using aggregate functions
-#' such as `sum()` and `mean()`.
+#' are included at least once in the table, as well as the values of any `by` variables.
+#'
+#' Additionally, filters can be applied on individual column values (if a `by` variable was specified) via the `n_XX`,
+#' `N_XX`, and `p_XX` statistics, where each `XX` represents the index of the column to select the statistic from. For
+#' example, `filter = n_1 > 5` will check whether `n` values in the first column of the table are greater than 5 in each
+#' row.
+#'
+#' Overall statistics for each row can be used in filters via the `n_overall`, `N_overall`, and `p_overall` statistics.
+#' If used in `filter`, overall statistics are derived within the filtering function. `n_overall` can only be derived if
+#' `n` statistic is present in the table for the filter variable, `N_overall` if the `N` statistic is present for the
+#' filter variable, and `p_overall` if both the `n` and `N` statistics are present for the filter variable.
+#'
+#' By default, filters will be applied at the level of the innermost hierarchy variable, i.e. the last variable supplied
+#' to `variables`. If filters should instead be applied at the level of one of the outer hierarchy variables, the `var`
+#' parameter can be used to specify a different variable to filter on. When `var` is set to a different (outer) variable
+#' and a level of the variable does not meet the filtering criteria then the section corresponding to that variable
+#' level - including summary rows - and all sub-sections within that section will be removed.
+#'
+#' In addition to filtering on individual statistic values, filters can be applied across the row (i.e. across all `by`
+#' variable values) by using aggregate functions such as `sum()` and `mean()`. For simplicity, it is suggested to use
+#' the `XX_overall` statistics in place of `sum(XX)` in equivalent scenarios. For example, `n_overall` is equivalent to
+#' `sum(n)`.
 #'
 #' If an overall column was added to the table (via `add_overall())`) this column will not be used in any filters (i.e.
-#' `sum(n)` will not include the overall `n` in a given row). To filter on overall statistics use the `sum()` function
-#' in your filter instead (i.e. `sum(n)` is equal to the overall column `n` across any `by` variables).
+#' `n_overall` will not include the overall `n` in a given row).
```

Comment:
I think with the addition of `n_overall`, this sentence is a little confusing. Maybe we can exclude it now?